Hestia/Hestia/frontend/EntryScreen/UIViewControllerLocalGlobal.cs

```diff
@@ -0,0 +1,217 @@
+using Foundation;
+using System;
+using System.Collections.Generic;
+using UIKit;
+using Hestia.backend.utils;
+using Hestia.backend;
+using Hestia.Resources;
+using Hestia.DevicesScreen.resources;
+using Auth0.OidcClient;
+using Hestia.backend.authentication;
+using System.Threading.Tasks;
+using IdentityModel.OidcClient;
+using System.Diagnostics;
+using Hestia.backend.exceptions;
+
+namespace Hestia
+{   // This view controller belongs to the first window that can be shown on loading the app
+    // if no user default for local/global is present. The user can then choose local/global.
+    public partial class UIViewControllerLocalGlobal : UIViewController
+    {
+        Auth0Client client;
+
+        // User defaults
+        NSUserDefaults userDefaults;
+        string defaultServerName;
+        string defaultIP;
+        string defaultPort;
+        string defaultAccessToken;
+        string defaultIdentityToken;
+
+        public UIViewControllerLocalGlobal (IntPtr handle) : base (handle)
+        {
+        }
+
+        public override void ViewDidLoad()
+        {
+            base.ViewDidLoad();
+            userDefaults = NSUserDefaults.StandardUserDefaults;
+            defaultServerName = userDefaults.StringForKey(strings.defaultsServerNameHestia);
+            defaultIP = userDefaults.StringForKey(strings.defaultsIpHestia);
+            defaultPort = userDefaults.StringForKey(strings.defaultsPortHestia);
+            defaultAccessToken = userDefaults.StringForKey(strings.defaultsAccessTokenHestia);
+            defaultIdentityToken = userDefaults.StringForKey(strings.defaultsIdentityTokenHestia);
+
+            // Already anticipate local login
+            // Check if local serverinformation is present and correct
+            bool validIp = CheckLocalLoginDefaults();
+
+            ToLocalButton.TouchUpInside += delegate (object sender, EventArgs e)
+            {
+                userDefaults.SetString(bool.TrueString, strings.defaultsLocalHestia);
+                Globals.LocalLogin = true;
+
+                if (validIp)
+                {
+                    SetValuesAndSegueToDevicesMainLocal();
+                }
+                else
+                {
+                    Console.WriteLine("To serverconnect");
+                    PerformSegue(strings.mainToServerConnect, this);
+                }
+            };
+
+            ToGlobalButton.TouchUpInside += async delegate (object sender, EventArgs e)
+            {
+                if (HasValidGlobalLogin())
+                {
+					SetValuesAndSegueToServerSelectGlobal();

```

Comment:
weird indentation

Reply:
Hmm was not visible in Visual Studio, but I think it is fixed in a8201d215aed3690f28a8903a1af3333351ae246